web/src/user_group_edit.ts

```diff
@@ -1533,28 +1557,94 @@ export function maybe_reset_right_panel(groups_list_data: UserGroup[]): void {
     }
 }
 
+function realm_has_deactivated_user_groups(): boolean {
+    const realm_user_groups = user_groups.get_realm_user_groups(true);
+    const deactivated_group_count = realm_user_groups.filter((group) => group.deactivated).length;
+
+    return deactivated_group_count > 0;
+}
+
+function user_subscribed_to_deactivated_user_groups(): boolean {
+    const groups = user_groups.get_user_groups_of_user(people.my_current_user_id(), true);
+    const deactivated_group_count = groups.filter((group) => group.deactivated).length;
+    return deactivated_group_count > 0;
+}
+
 export function update_empty_left_panel_message(): void {
     // Check if we have any groups in panel to decide whether to
     // display a notice.
-    let has_groups;
     const is_your_groups_tab_active =
         get_active_data().$tabs.first().attr("data-tab-key") === "your-groups";
+
+    let has_groups;
+    const current_group_filter =
+        filters_dropdown_widget.value() ?? FILTERS.ACTIVE_AND_DEACTIVATED_GROUPS;
+    const include_deactivated_groups =
+        current_group_filter === FILTERS.DEACTIVATED_GROUPS ||
+        current_group_filter === FILTERS.ACTIVE_AND_DEACTIVATED_GROUPS;
     if (is_your_groups_tab_active) {
-        has_groups = user_groups.get_user_groups_of_user(people.my_current_user_id(), true).length;
+        const groups = user_groups.get_user_groups_of_user(
+            people.my_current_user_id(),
+            include_deactivated_groups,
+        );
+        if (current_group_filter === FILTERS.DEACTIVATED_GROUPS) {
+            has_groups = user_subscribed_to_deactivated_user_groups();
+        } else {
+            has_groups = groups.length;
+        }
     } else {
-        has_groups = user_groups.get_realm_user_groups().length;
+        const realm_user_groups = user_groups.get_realm_user_groups(include_deactivated_groups);
+
+        if (current_group_filter === FILTERS.DEACTIVATED_GROUPS) {
+            has_groups = realm_has_deactivated_user_groups();
+        } else {
+            has_groups = realm_user_groups.length;
```

Comment:
I think this will be more readable if conditions are like this -
```ts
let groups;
if (is_your_groups_tab_active) {
    groups = get_realm_user_groups(true);
} else {
    groups = get_user_groups_of_user(user_id, true)
}

if (filter === ACTIVE_AND_DEACTIVATED_GROUPS) {
    has_groups = groups.length > 0;
} else  if (filter === ACTIVE_GROUPS) {
    has_groups = groups.filter((group) => !group.deactivated).length > 0;
} else {
    has_groups = groups.filter((group) => group.deactivated).length > 0;
}
```